pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -977,6 +939,9 @@ func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1beta
 				in corruption or other issues for applications with singleton requirement. After the host is powered
 				off we know for sure that it is safe to re-assign that workload to other nodes.
 			*/
+			if err := a.storeAnnotationsAndLabels(ctx, node, machine); err != nil {
```

Comment:
WDYT about requeue here even if there's no error?
This is how the rest of the code works here (do one thing, requeue)

something like:
```
if machine.Annotations[nodeAnnotationsBackupAnnotation] != marshaledAnnotations ||
		machine.Annotations[nodeLabelsBackupAnnotation] != marshaledLabels {
        return  a.storeAnnotationsAndLabels(ctx, node, machine)
}
```

TBH I've been using that approach after reading [your comment](https://github.com/metal3-io/cluster-api-provider-baremetal/pull/225#discussion_r369349870) a while ago.
I hope I interpreted it correctly